ir/memory.cpp

```diff
@@ -365,12 +365,16 @@ unsigned Pointer::total_bits() {
 expr Pointer::is_local() const {
   if (num_locals == 0)
     return false;
-  if (num_nonlocals == 0)
+  if (m.num_nonlocals() == 0)
     return true;
   auto bit = total_bits() - 1;
   return p.extract(bit, bit) == 1;
 }
 
+expr Pointer::nonlocal_wf() const {
```

Comment:
wf stands for wellformedness

Reply:
I don't really like this function because it only makes sense for non_local ptrs. It's wrong to call it otherwise.
We can't assert or do something else because it's context sensitive. I would rollback this change; it's unrelated with this PR.

Reply:
I thought it makes sense because otherwise repeated changes on ule(num_nonlocal) were needed.
Also it is natural to have a function for encoding wellformedness of a pointer; local pointers don't need wellformedness right now, that was why nonlocal pointers were only considered.

The change seemed orthogonal with this PR so I rollbacked it, but I think it will be great if the refactoring is accepted before I work on #284 ; I have an idea for #284, but it will again make repeated changes regarding num_nonlocal.